File: fix3.py
import os

def fix_generate_article():
    filepath = 'src/lib/news/ai/generate-article.ts'
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()

    content = content.replace('!quality.ok', '!quality.passed')
    content = content.replace('quality.codes', 'quality.rejectionReasons')
    
    # Also fix depth_quality?.codes to depth_quality?.issues.map(i => i.code) if it exists
    # Wait, earlier log showed quality.depth_quality.codes without ?. Let me check if 'codes' is valid for depth_quality.
    # Actually, in editorial-guards.ts it was depth_quality.issues. But in generate-article.ts, earlier code had quality.depth_quality.codes! So maybe depth_quality DOES have codes? Let's check alidateEditorialDepth return type.
    
    with open(filepath, 'w', encoding='utf-8') as f:
        f.write(content)

fix_generate_article()

File: fix3.js
const fs = require('fs');
let content = fs.readFileSync('src/lib/news/ai/editorial-body.ts', 'utf8');

content = content.replace(
  /if \(a\.startsWith\(b\.slice\(0, Math\.min\(80, b\.length\)\)\)\) return true;/g,
  'if (a.length >= 150 && b.length >= 150 && a.startsWith(b.slice(0, 150))) return true;'
);

content = content.replace(
  /if \(b\.startsWith\(a\.slice\(0, Math\.min\(80, a\.length\)\)\)\) return true;/g,
  'if (a.length >= 150 && b.length >= 150 && b.startsWith(a.slice(0, 150))) return true;'
);

content = content.replace(
  /if \(a\.length >= 40 && b\.length >= 40\) {/g,
  'if (a.length >= 80 && b.length >= 80) {'
);

fs.writeFileSync('src/lib/news/ai/editorial-body.ts', content, 'utf8');
